packages/data-transport-layer/src/services/main/service.ts

```diff
@@ -77,13 +85,150 @@ export class L1DataTransportService extends BaseService<L1DataTransportServiceOp
+        let aList
+        try {
+          aList = JSON.parse(await this.state.db.get("omgx-addr"))
+        } catch(e) {
+          if (e.notFound) {
+            this.logger.warn("Address Registry is not yet ready to serve OMGX addresses (db notFound)")
+            return res.status(503).json({error: "Address Registry is not yet populated"})
+          } else { throw e }
+        }
+
+        return res.json(aList)
+      } catch (e) {
+        return res.status(400).json({
+          error: e.toString(),
+        })
+      }
+    })
+    this.state.addressRegistry['put']("/addresses.json", async (req, res) => {
+      try {
+        const rb = req.body
+
+        this.logger.info("addressRegistry PUT request for base addresses", {rb})
+
+        let addrList = {}
+
+        /* Future - if there's an existing list, compare it with the new one and reject
+           any attempts to change certain critical addresses. For now, only cares about
+           AddressManager since that's used directly by dtl */
+
+        try {
+          const _addrList = await this.state.db.get("address-list")
+          addrList = JSON.parse(_addrList)
+
+          if (rb['AddressManager'] && addrList['AddressManager'] && rb['AddressManager'] !== addrList['AddressManager']) {
+            this.logger.error("Can't overwrite saved addressManager value", { old: addrList['AddressManager'], new:rb['AddressManager']})
+            return res.status(400).json({error:"Can't overwrite saved AddressManager value"})
+          }
+        } catch(e) {
+          if (e.notFound) {
+            this.logger.info("No previous address list was found")
+          } else { throw e; }
+        }
+
+        this.logger.info("MMDBG Will store", rb)
+        await this.state.db.put("address-list", JSON.stringify(rb))
+        this.logger.info("MMDBG Stored")
+        return res.sendStatus(201).end()
+      } catch (e) {
+        return res.status(400).json({
+          error: e.toString(),
+        })
+      }
+    })
+    this.state.addressRegistry['put']("/omgx-addr.json", async (req, res) => {
+      try {
+        const rb = req.body
+
+        this.logger.info("addressRegistry PUT request for OMGX addresses", {rb})
+
+        // As with the base list, we could add future restrictions on changing
+        // certain critical addresses. For now we allow anything.
+
+        this.logger.info("MMDBG Will store", rb)
+        await this.state.db.put("omgx-addr", JSON.stringify(rb))
+        this.logger.info("MMDBG Stored")
+        return res.sendStatus(201).end()
+      } catch (e) {
+        return res.status(400).json({
+          error: e.toString(),
+        })
+      }
+    })
+
+    this.state.addressRegistry['get']("/state-dump.latest.json", async (req, res) => {
+      try {
+        //const sd = await this.state.db.get("state-dump")
+        //return res.send()
+        return res.sendFile("/opt/optimism/packages/data-transport-layer/state-dumps/state-dump.latest.json")
```

Comment:
Not planning on running this outside of docker?

Reply:
I don't think we have any immediate plans to run it outside of docker. The path could be made configurable, but at that point I'd also want to explore other options for handling the state-dump file (e.g. uploading it at runtime with 'curl', or putting it into the l2geth docker image). Putting it into DTL like this was just the quickest way of removing our 'deployer' dependency.